docs/secrets/Secrets.md

```diff
@@ -1,98 +1,217 @@
+Details can be found on the [Examples page][jenkins_k8s_credential_provider_plugin_examples] of the plugin.
 
-### Jenkinsfile Fetch Secret
+When a pipeline gets executed in a transient sandbox namespace, the secrets listed in `spec.secrets` of the corresponding PipelineRun resource object are copied to the sandbox namespace with the same name.
+The Jenkins Kubernetes Credentials Provider Plugin will use the secrets from the sandbox namespace only.
+Any secret that is not listed in `spec.secrets` will not be available as Jenkins credential.
 
-We need credentials to fetch the Jenkinsfile, before we execute it.
-The fetch is done inside our Jenkinsfile Runner image in the start script. The Git credentials are [injected by Tekton](https://github.com/tektoncd/pipeline/blob/master/docs/auth.md)) and the script can simply do a `git clone` without prior login.
+__:warning: Warning:__ Any code that gets executed by a pipeline AND has access to the Kubernetes service account token can read all image pull secrets! This is especially important to consider if untrusted code may get executed, e.g. a pipeline processing pull requests from untrusted users.
 
-As first step we only plan to support our predefined Jenkinsfile. But in future teams will be able to use custom pipelines, either from the project sources or another custom git repo.
+- Any code running in the Jenkinsfile Runner container, because this container has the service account token mounted.
+- Any code running in additional containers that have the service account token mounted.
 
-- **Centrally defined Jenkinsfile**: The credentials have to be provided by us.
-- **Custom Jenkinsfile**: Credentials have to be provided by customer
-    - **In Sources**: Same as [Project Source Fetch Secrets]() (see below)
-    - **Separate Repo**: Separate Credentials
+To prevent access to secrets, untrusted code must be executed in containers where the service account token will not be supplied to (mounting of service account token disabled via pod spec and token not passed into the container in any other way).
 
-**Personas:** `CloudCi` & `Customer`
 
+## Other Secrets
 
-### Source Project Secret
+### Log Storage in ElasticSearch
 
-Once the pipeline is running it needs to sync the project sources. This secret might also require write permissions to the repository, e.g. to create a tag, send status feedback, push an updated version commit.
+The log output of pipeline runs can be send to an Elasticsearch server.
+The credentials to authenticate at the Elasticsearch server must be provided by the Steward client.
 
-The required credentials can be provided by customers as [General Jenkins Secrets]() (see below).
+See documenation page [Sending Pipeline Logs to Elasticsearch](../pipeline-logs-elasticsearch/README.md).
 
-**Personas:** `Customer`
+__TODO:__ How to configure credentials for Elasticseach logging
```

Comment:
Open TODO

Reply:
Intended to be fixed later, as we don't have a working solution yet.